### PR TITLE
Add entity prototype save test

### PR DESCRIPTION
**Pull request by ElectroJr**
This adds a new test that is intended to ensure that when an entity prototype is spawned into an uninitialized map and then that map gets saved, that the resulting entity is effectively just the prototype. I.e., the entity shouldn't add new components and it shouldn't modify any component data.

This helps by:
- reducing map file sizes
- reduces the number of dirty components that need to get sent to clients. 
- the test will generally error if people use component-init to perform randomization & other one-time set up that should be done on map init (e.g., spawning entities).

This is also somewhat related to #8423. This is currently a draft, because there are many entities that need fixing. Currently mobs and the singularity are exempt from the test, because they aren't map-saveable.

As an example, the  output from the test looks like this:
```
14) Prototype CrateFoodKvassTank gains a component on spawn: ItemSlots
15) Prototype CableTerminal gains a component on spawn: CollideOnAnchor
16) Prototype BlastDoor gains a component on spawn: SignalReceiver
17) Prototype BlastDoorOpen modifies component on spawn: Occluder. Modified fields: enabled
18) Prototype BlastDoorOpen modifies component on spawn: Physics. Modified fields: canCollide
19) Prototype BlastDoorOpen modifies component on spawn: Airtight. Modified fields: airBlocked
20) Prototype BlastDoorOpen gains a component on spawn: SignalReceiver
21) Prototype WeaponWandPolymorphBase modifies component on spawn: BasicEntityAmmoProvider. Modified fields: count
```

There are still a bunch of prototypes that fail the test. But the PR is already quite sizeable, and the number of failing entities keeps increasing every now and then.

So for now I've just thrown the remainder in an ignore list, and will create a separate issue indicating that these specific prototypes need to get fixed.

### CODE REVIEW

**Comment by metalgearsloth** [comment]
collisionwake + ambience + containers are going to be a lot of these I think.

**Comment by mirrorcult** [comment]
finally getting rid of the init ensurecomp cruft are we

**Comment by ElectroJr** [comment]
Well starting on it. Something like the `[RequiresComponent(typeof(blah))]` would still be nice. 

> collisionwake + ambience + containers are going to be a lot of these I think.

I know a couple hundred are just because the base drink proto is missing `DrainableSolution` & `RefillableSolution`, and a bunch are just because of base storage missing `ContainerContainer`. Fixing 90% of the issues will probably be easy and just involve tweaking base prototypes, the remainder will probably be quite annoying.

**Comment by ElectroJr** [comment]
I'm tempted to just chuck the rest of the erroring prototypes into the test's list of ignored prototypes and fix them later in smaller PRs. This PR already changes so much yaml & will attract a lot of conflicts.

File: Resources/Prototypes/Entities/Structures/Dispensers/chem.yml [hunk 1]
```diff
@@ -12,9 +12,6 @@
     emagPack: ChemDispenserEmaggedInventory
   - type: ApcPowerReceiver
   - type: ExtensionCableReceiver
-  - type: Construction
-    graph: Machine
-    node: machine
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Structures/Dispensers/chem.yml:17]: removed with nothing else added?

**Review by mirrorcult — APPROVED** [review]

File: Resources/Prototypes/Entities/Structures/Dispensers/chem.yml (hunk 1 above)
Comment (ElectroJr) [reply Resources/Prototypes/Entities/Structures/Dispensers/chem.yml:17]: The dispenser already inherits from ReagentDispenserBase, which now newly inherits from ConstructibleMachine which adds this & some other missing construction related stuff.

**Pull request merged**